buildpackages: openstack image names not unique.

robbat2 opened the pull request:
OpenStack image names are not always unique, as noted by teuthology
commit d4c98512: openstack: image names only unique within tenants

Use the unique image ID to create the pkg host and the build hosts
instead (server create fails otherwise).

Signed-off-by: Robin H. Johnson robin.johnson@dreamhost.com

Commit:
buildpackages: openstack image names not unique.

OpenStack image names are not always unique, as noted by teuthology
commit d4c98512: openstack: image names only unique within tenants

Use the unique image ID to create the pkg host and the build hosts
instead (server create fails otherwise).

Signed-off-by: Robin H. Johnson <robin.johnson@dreamhost.com>

dachary commented on tasks/buildpackages/Makefile:
```diff
@@ -30,7 +37,7 @@ ${HOME}/.ssh_agent:
 
 flock-${PKG_REPO}:
-	openstack server create --image 'teuthology-ubuntu-14.04' --flavor ${HTTP_FLAVOR} --key-name teuthology --security-group teuthology --property ownedby=${MY_IP} --user-data ${PKG_REPO_USER_DATA} --wait ${PKG_REPO}
+	openstack server create --image ${IMAGE_ID} --flavor ${HTTP_FLAVOR} --key-name teuthology --security-group teuthology --property ownedby=${MY_IP} --user-data ${PKG_REPO_USER_DATA} --wait ${PKG_REPO}
```
this is going to incorrectly map to teuthology-centos-7.0 if called to build a centos-7.0 package, it needs to be teuthology-ubuntu-14.04 instead

robbat2 replied:
Updated PR

Commit:
buildpackages: catch VM instances in ERROR state.

It is possible for 'server create' to return success, but then NOT have
a good VM afterwards. The instance will be in state ERROR.

Signed-off-by: Robin H. Johnson <robin.johnson@dreamhost.com>

Commit:
buildpackages: fix configure on Ubuntu 12.04.

Ubuntu 12.04/precise has a buggy lttng; it fails our lttng-is-sane
check, as it leaves out headers.

This should not stop building of Ceph, as the automatic build handling
from debian/rules will correctly omit LTTNG on Precise.

Signed-off-by: Robin H. Johnson <robin.johnson@dreamhost.com>

Commit:
buildpackages: support ubuntu 12.04.

Signed-off-by: Robin H. Johnson <robin.johnson@dreamhost.com>

Commit:
buildpackages: more flexibility on lock filename for build concurrency.

Signed-off-by: Robin H. Johnson <robin.johnson@dreamhost.com>

Commit:
buildpackages: add build timeouts.

The build/upload process CAN get stuck sometimes. As we are paying
utility rates for the high-power build server, we should timeout when
it's likely the build is stuck.

Use a default timeout just short of 4 hours, as some providers round up
partial instance-hours for billing.

Signed-off-by: Robin H. Johnson <robin.johnson@dreamhost.com>
